libs/metagen/src/mdk/mdk.wit

```diff
@@ -0,0 +1,42 @@
+package metatype:mdk;
+
+interface shared {
+  record req {
+    op-name: string,
+    in-json: string,
+  }
```

Comment:
The way I am thinking of this is something like..
```rust
fn handle(req: Req) {
   // deserialize to a known *generated* type
   let inp: Struct = deserialize(req.in_json..);
   let out = match req.op {
       "add" => inp.a + inp.b,
       "sub" => inp.a - inp.b,
        _ => { return Err(...) }
   };

  // serialize result to a known *generated* type
   Ok(serialize::<Integer>(out))
}
```
is this correct?

Reply:
Although `op_name`  confuses me a bit since a materializer maps to a single op.

Reply:
Hey, right on with the comment on `op_name`. I POCed that with the old system in mind and now that we're generating the interface ourselves, we can probably find other ways of mapping functions besides manually setting the name.

Reply:
For example, we can just map using the function type id. Ideally tho, the interface shouldn't assume we've done codegen. The interface should assume anything about codegen.